src/hotspot/share/code/nmethod.cpp

```diff
@@ -470,9 +470,9 @@ nmethod* nmethod::new_native_nmethod(const methodHandle& method,
   code_buffer->finalize_oop_references(method);
   // create nmethod
   nmethod* nm = NULL;
+  int native_nmethod_size = CodeBlob::allocation_size(code_buffer, sizeof(nmethod));
```

Comment:
This change and another below are not calculation of stub sizes. You need either create another PR for them or update JDK-8289071 to include these cases.